umbraco/Umbraco-CMS
# V9: Allowlisting remote URLs for displaying content on the content dashboard

## Details
- Fixed ContentDashboard setting to be configurable from config file;
  - You can now change the values of both `AllowContentDashboardAccessToAllUsers` and `ContentDashboardPath` as in v8; 
- V9 version of #11822;
- Added the ability to allowlist the remote URLs for the "Getting Started" dashboard in config through new setting : `ContentDashboardUrlAllowlist`;
- If no addresses are specified, any address is allowed.

## Test
- Without specifying any addresses, the "Getting Started" dashboard should display the content as before.
- Add the following in appsettings.json and verify that the behaviour persists:
```json
{
  "Umbraco": {
    "CMS": {
      . . .
      "ContentDashboard": {
        "ContentDashboardUrlAllowlist": [
          "https://dashboard.umbraco.com/",
          "https://dashboard.umbraco.org/",
          "https://google.com/"
        ]
      },
      . . .
    }
  }
}
```
- Remove either https://dashboard.umbraco.com/ (and observe that the content is not fetched) or https://dashboard.umbraco.org/ (and observe that the CSS is not fetched) 
  - Check that the corresponding calls to the **GetRemoteDashboardContent** or **GetRemoteDashboardCss** endpoints return 400 and that the errors are also in the log (Settings > Log Viewer)
